autoload/codefmt.vim

```diff
@@ -186,19 +223,39 @@ function! codefmt#GetClangFormatFormatter() abort
       let l:cmd += ['-assume-filename', l:fname]
     endif
 
-    if empty(a:ranges)
-      return
-    endif
     for [l:startline, l:endline] in a:ranges
       call maktaba#ensure#IsNumber(l:startline)
       call maktaba#ensure#IsNumber(l:endline)
       let l:cmd += ['-lines', l:startline . ':' . l:endline]
     endfor
 
+    " Version 3.4 introduced support for cursor tracking
+    " http://llvm.org/releases/3.4/tools/clang/docs/ClangFormat.html
+    let l:supports_cursor = s:ClangFormatHasAtLeastVersion([3, 4])
+    if l:supports_cursor
+      " line2byte counts bytes from 1, and col counts from 1, so -2 
+      let l:cursor_pos = line2byte(line(".")) + col(".") - 2
+      let l:cmd += ['-cursor', string(l:cursor_pos)]
+    endif
+
     let l:input = join(getline(1, line('$')), "\n")
     let l:result = maktaba#syscall#Create(l:cmd).WithStdin(l:input).Call()
     let l:formatted = split(l:result.stdout, "\n")
-    call maktaba#buffer#Overwrite(1, line('$'), l:formatted)
+
+    if !l:supports_cursor
+      call maktaba#buffer#Overwrite(1, line('$'), l:formatted[0:])
+    else
+      call maktaba#buffer#Overwrite(1, line('$'), l:formatted[1:])
+      try
+        let l:clang_format_output_json = maktaba#json#Parse(l:formatted[0])
+        let l:new_cursor_pos =
+              \ maktaba#ensure#IsNumber(l:clang_format_output_json.Cursor) + 1
```

Comment:
Any scenarios where either no JSON gets returned or you get JSON w/o Cursor that you might want custom messages for?